specification.md

```diff
@@ -0,0 +1,92 @@
+will align with the compatibility requirements defined in
+this spec.
+
+## Goals
+
+* OpenTelemetry collector can be used as a drop-in replacement
+  for Prometheus server to scrape and export metrics data.
+* OpenTelemetry collector should export OTLP-compatible Prometheus
+  time series to OpenTelemetry metrics exporters.
+* OpenTelemetry libraries should implement exporters to publish
+  Prometheus metrics.
+* To the OpenTelemetry collector, an OpenTelemetry target and
+  a Prometheus instrumented target is indistinguishable.
+
+## Differences and Limitations
+
+OpenTelemetry and Prometheus/OpenMetrics have different design
+goals and this reflects in the data model and implementation
+details. This section summarizes a few key differences.
+
+* **Pull vs push**: Prometheus is mainly designed for pull
+  whereas OpenTelemetry primarily is designed for push. This
+  difference causes how the state is maintained throughout the
+  collection including how it’s handled in the OpenTelemetry
+  collector.
+* **Cumulative vs delta**: OpenTelemetry supports delta temporality
+  whereas Prometheus always expects absolute/cumulative values. This
+  breaks some components that produce and communicate deltas where
+  cumulatives cannot be rebuilt before being exported to Prometheus.
+* **Histogram boundaries**: Prometheus histogram boundaries are by
+  lower equal (le) while OpenTelemetry histogram boundaries are
+  greater equal (ge). (This difference will be resolved via #18)
+* **Semantic conventions**: OpenTelemetry predefines semantic
+  conventions to collect additional metadata with telemetry data.
+  Prometheus users don’t follow the same conventions and Prometheus
+  client library provided data may lack the semantic conventions
+  available in OpenTelemetry libraries.
+
+## Compatibility Requirements
+
+Given the number of fundamental design goals between OpenTelemetry
+and Prometheus, our aim is to close the gaps where possible and
+make the right compromises to meet the goals defined in this document.
+OpenTelemery and Prometheus won’t be fully compatible but we will
+enable important use cases to enable OpenTelemetry for Prometheus
+users. The following sections summarizes the expectations from
+the collector and the libraries.
+
+### Collector
+
+* Collector will implement a Prometheus remote write exporter.
+  Publishing a pull-based metrics handler with all collected
+  metrics is not a scalable approach.
+* Collector will support scraping and ingesting cumulative metrics.
+  Prometheus doesn’t support deltas and there are cases where
+  rebuilding the cumulatives from deltas is not possible/easy.
+* Collector will support all discovery and scraping configuration
+  options in the Prometheus server. Collector will ignore the
+  alerting rules.
+* Collector will support exporting cumulative series to
+  OTLP-compatible exporters.
+* At each scrape, collector will produce an "up" metric as a gauge
```

Comment:
I will pursue the proposal discussed in this [thread](https://github.com/open-telemetry/opentelemetry-specification/issues/1078#issuecomment-776404440), where the receiver will attach both "alive" and "present" metrics. The PRW exporter would translate "alive" and "present" into "up" and this will be easy for the Prometheus-to-PRW code path because points will be aligned by the receiver. 

The reason for my proposal is that we can push OTLP data with just the "alive" metric, and join it with "present" from service discovery, producing aligned data that can be exported through PRW in the same form, yielding an "up" metric in PRW from pushed OTLP data combined with service discovery data.

Reply:
How often will "present" be sent, and how do you ensure its timing is aligned with "alive"? I seem to remember a discussion about pitfalls and corner cases with a similar approach from some time back. I will try to find it and post if it's useful here.